Lib/configparser.py

```diff
@@ -692,6 +700,10 @@ def add_section(self, section):
         if section == self.default_section:
             raise ValueError('Invalid section name: %r' % section)
 
+        if section is UNNAMED_SECTION:
+            if not self._allow_unnamed_section:
+                raise UnnamedSectionDisabledError()
```

Comment:
I suggested a new exception because there was the `DuplicateSectionError` that existed, so I wasn't sure whether another exception would have been relevant.

Reply:
Yes, I agree that this was the best option.